core/src/main/java/org/mule/routing/MessageFilter.java

```diff
@@ -90,7 +90,7 @@ protected boolean accept(MuleEvent event)
 
         if (event != null && !VoidMuleEvent.getInstance().equals(event))
```

Comment:
Is related as you added another method to filter an event and deprecating the other one, which I think must be removed instead